test/Microsoft.Extensions.Internal.Test/TypeNameHelperTest.cs

```diff
@@ -9,110 +9,124 @@ namespace Microsoft.Extensions.Internal
 {
     public class TypeNameHelperTest
     {
-        [Fact]
-        public void Can_pretty_print_CLR_full_name()
+        public static IEnumerable<object[]> GetFullTypeNamesTestData()
         {
             // Predefined Types
-            Assert.Equal("int",
-                TypeNameHelper.GetTypeDisplayName(typeof(int)));
-            Assert.Equal("System.Collections.Generic.List<int>",
-                TypeNameHelper.GetTypeDisplayName(typeof(List<int>)));
-            Assert.Equal("System.Collections.Generic.Dictionary<int, string>",
-                TypeNameHelper.GetTypeDisplayName(typeof(Dictionary<int, string>)));
-            Assert.Equal("System.Collections.Generic.Dictionary<int, System.Collections.Generic.List<string>>",
-                TypeNameHelper.GetTypeDisplayName(typeof(Dictionary<int, List<string>>)));
-            Assert.Equal("System.Collections.Generic.List<System.Collections.Generic.List<string>>",
-                TypeNameHelper.GetTypeDisplayName(typeof(List<List<string>>)));
+            yield return new object[] { "int", typeof(int) };
```

Comment:
More of a nit, you could return `TheoryData` instead. It removes some of the cruft around declaring these:

```C#
public static TheoryData GetFullTypeNamesTestData()
{ 
   return new TheoryData<Type, string>
   {
      { typeof(int), "int" },
      ...
    };
}
```

Also generally, inputs appear first in the test method arg list followed by expectations.

Reply:
Thank you, I did know about that, look much better with concrete types and without arrays of objects.